metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -499,7 +500,10 @@ final case class Indexer(
       targetOpt: Option[b.BuildTargetIdentifier],
       data: Seq[TargetData],
   ): Unit = {
-
+    def isExtensionMethod(si: SymbolInformation): Boolean =
+      si.kind == Kind.METHOD && si.properties == 0 && si.symbol(
```

Comment:
You could potentially use the descriptor parser for the second part of the condition. Take a look at DefinitionAlternatives for examples

Reply:
I wish to get rid of those hacky conditions, which work only for ScalaToplevelMtags not in general. When we produce extension methods in  ScalaToplevelMtags they don't have a "disambiguator" added (I don't even know why) and that's the property used here. I know, that @tanishiking made a feature request about adding is extension method information to SymbolInformation but nothing was done so far about this. Maybe one of you has a better way to mark extension methods, which is a bit cleaner?

Reply:
Since `.properties` is an Int we could potentially just use an unused bit such as:
```
import scala.meta.internal.semanticdb.SymbolInformation.{Property => p}
// needs to be put somewhere accessible
val EXTENSION = p.values.last.value << 1

implicit class XtensionSemanticdbSymbolInformation(private val info: SymbolInformation){
   def isExtension = EXTENSION & info.properties != 0
}
```

EXTENSION property can be added when creating the extension method since it doesn't really have any other properties.

This can be later updated when PR for scalameta is merged.

Reply:
Great. I wasn't sure I can do that, that there are no checks or internal usages for it.

Reply:
We could even do `val EXTENSION = p.values.map(_.value).max << 1` which would mean that would never have an issue with a real property.